python/tests/graph/test_input_graph.py

```diff
@@ -0,0 +1,223 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import absolute_import, division, print_function
+
+from contextlib import contextmanager
+from glob import glob
+import os
+import shutil
+import tempfile
+
+import numpy as np
+import tensorflow as tf
+
+from sparkdl.graph.input import *
+import sparkdl.graph.utils as tfx
+
+from ..tests import PythonUnitTestCase
+
+
+class TFInputGraphTest(PythonUnitTestCase):
+
+    def setUp(self):
+        self.vec_size = 23
+        self.num_samples = 107
+
+        self.input_col = 'dfInputCol'
+        self.input_op_name = 'tnsrOpIn'
+        self.output_col = 'dfOutputCol'
+        self.output_op_name = 'tnsrOpOut'
+
+        self.feed_names = []
+        self.fetch_names = []
+        self.input_mapping = {}
+        self.output_mapping = {}
+        self.setup_iomap(replica=1)
+
+        self.input_graphs = []
+        self.test_case_results = []
+        # Build a temporary directory, which might or might not be used by the test
+        self.model_output_root = tempfile.mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self.model_output_root, ignore_errors=True)
+
+    def setup_iomap(self, replica=1):
+        self.input_mapping = {}
+        self.feed_names = []
+        self.output_mapping = {}
+        self.fetch_names = []
+
+        if replica > 1:
+            for i in range(replica):
+                colname = '{}_replica{:03d}'.format(self.input_col, i)
+                tnsr_op_name = '{}_replica{:03d}'.format(self.input_op_name, i)
+                self.input_mapping[colname] = tnsr_op_name
+                self.feed_names.append(tnsr_op_name + ':0')
+
+                colname = '{}_replica{:03d}'.format(self.output_col, i)
+                tnsr_op_name = '{}_replica{:03d}'.format(self.output_op_name, i)
+                self.output_mapping[tnsr_op_name] = colname
+                self.fetch_names.append(tnsr_op_name + ':0')
+        else:
+            self.input_mapping = {self.input_col: self.input_op_name}
+            self.feed_names = [self.input_op_name + ':0']
+            self.output_mapping = {self.output_op_name: self.output_col}
+            self.fetch_names = [self.output_op_name + ':0']
+
+    @contextmanager
+    def _run_test_in_tf_session(self):
+        """ [THIS IS NOT A TEST]: encapsulate general test workflow """
+
+        # Build the TensorFlow graph
+        graph = tf.Graph()
+        with tf.Session(graph=graph) as sess, graph.as_default():
+            # Build test graph and transformers from here
+            yield sess
+
+            ref_feed = tfx.get_tensor(graph, self.input_op_name)
+            ref_fetch = tfx.get_tensor(graph, self.output_op_name)
+
+            def check_input_graph(tgt_gdef, test_idx):
```

Comment:
just put the internals of the function inside the for-loop below since the function is not used anywhere else

Reply:
Well, it is another layer of indentation and it is hard to track which is aligned to which outside an editor.